src/client/BaseApi.ts

```diff
@@ -189,6 +231,31 @@ export class BaseApi implements IAstarApi {
         }
     }
 
+    public async getStakerInfo(address: string): Promise<bigint> {
+        await this.ensureConnection();
+
+        const [state, result] = await Promise.all([
+            this._api.query.dappStaking.activeProtocolState<PalletDappStakingV3ProtocolState>(),
+            this._api.query.dappStaking.stakerInfo.entries<Option<PalletDappStakingV3StakeAmount>>(address),
```

Comment:
Code became complicated (`reduce` call) just because you didn't use `ledger` instead `stakerInfo`. We discussed that and I explained that `ledger` has all stakes summed up.
Reduce call and a lot of logic can be omitted if you use `ledger`

Reply:
The reason why I choose this method is because I can foresee the next requests coming.

Someone will want to know for a specific dapp, and with this method I can easily filter it and only show that amount.
Or I can show amount per dapps staked.

Reply:
It makes sense, than leave as is